src/bika/lims/api/__init__.py

```diff
@@ -952,7 +954,15 @@ def get_path(brain_or_object):
     :rtype: string
     """
     if is_brain(brain_or_object):
-        return brain_or_object.getPath()
+        path = brain_or_object.getPath()
+        # Handle uid_catalog brains that may return relative paths
+        # Check if path is missing the portal path prefix
+        portal = get_portal()
+        portal_path = "/".join(portal.getPhysicalPath())
+        if not path.startswith(portal_path):
```

Comment:
Truly not expensive, but given the fact that this is called for every single brain and we might do this a lot of times (e.g. in listings via `get_url`), do you think it would be worth to add a ram cache `api.get_portal_path` function?

Reply:
In my opinion it is not worth it and I want to keep the API functions when possible without caching to avoid any side-effects.